src/map/script.c

```diff
@@ -8755,67 +8755,82 @@ static BUILDIN(grouprandomitem)
 }
 
 /*==========================================
- *
+ * makeitem <item_id>, <amount>, "<map name>", <X>, <Y> {, <showdropeffect>}};
  *------------------------------------------*/
 static BUILDIN(makeitem)
 {
-	int nameid,amount;
-	int x,y,m;
-	const char *mapname;
-	struct item item_tmp;
+	int nameid;
 
-	if( script_isstringtype(st, 2) ) {
-		const char *name = script_getstr(st, 2);
+	if (script_isstringtype(st, 2)) {
+		const char* name = script_getstr(st, 2);
 		struct item_data *item_data = itemdb->search_name(name);
 		if (item_data)
 			nameid = item_data->nameid;
 		else
 			nameid = UNKNOWN_ITEM_ID;
 	} else {
 		nameid = script_getnum(st, 2);
-		if( nameid <= 0 || !itemdb->exists(nameid)) {
-			ShowError("makeitem: Nonexistant item %d requested.\n", nameid);
+		if (nameid <= 0 || !itemdb->exists(nameid)) {
+			ShowError("buildin_makeitem: invalid item_id '%d'.\n", nameid);
 			return false; //No item created.
 		}
 	}
-	amount  = script_getnum(st,3);
-	mapname = script_getstr(st,4);
-	x       = script_getnum(st,5);
-	y       = script_getnum(st,6);
 
-	if(strcmp(mapname,"this")==0) {
-		struct map_session_data *sd = script->rid2sd(st);
+	int amount = script_getnum(st, 3);
+	const char *mapname = script_getstr(st, 4);
+	int16 x = script_getnum(st, 5);
+	int16 y = script_getnum(st, 6);
```

Comment:
It's better for these to be int, so that they can't underflow at runtime if you pass a number larger than 32k

```suggestion
	int x = script_getnum(st, 5);
	int y = script_getnum(st, 6);
```